packages/wrangler/src/dev-registry.tsx

```diff
@@ -83,6 +83,17 @@ export async function startWorkerRegistry() {
 		server = http.createServer(app);
 		terminator = createHttpTerminator({ server });
 		server.listen(DEV_REGISTRY_PORT);
+
+		/**
+		 * The registry server may have already been started by another wrangler process.
+		 * If wrangler processes are run in parallel, isPortAvailable() can return true
+		 * while another process spins up the server
+		 */
+		server.once('error', (err) => {
+			if ((err as unknown as { code: string }).code !== "EADDRINUSE") {
+				throw err;
+			}
+		});
```

Comment:
It's unclear to me how this works? Just from looking, it doesn't look like it'll retry with a different port. Would it be possible to add some retry logic here?

Reply:
It wouldn't work to retry with a different port, your worker registry port is hard-coded. The fix here is to not start the server, as it doesn't need to start (it's already running) and catch the EADDRINUSE instead of crashing wrangler.

It's probably not an optimal fix, I guess an improved isPortAvailable() check would be ideal, but I don't know how to ensure that returns the correct value when starting in parallel.

I've set up a turborepo project with three workers using wrangler init which replicates the issue.
https://github.com/willtemperley/wrangler2-issue-2027

Reply:
The dev registry is supposed to "just exist" on a well-known port, and spin up a new one at the same port if one doesn't exist. Issue occurs when you're trying to spin up multiple wranglers at the same time - will give this a spin tomorrow if @caass doesn't get a chance